### PR TITLE
Small tweaks to the recursion walk we do when looking for strign indentation regions

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Tiny thing i noticed while looking at traces:

image

Will also be helped by https://github.com/dotnet/roslyn/pull/73650

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@ToddGrun ptal.

File: src/Features/CSharp/Portable/StringIndentation/CSharpStringIndentationService.cs [hunk 1]
```diff
@@ -19,69 +20,62 @@
...
+        using var result = TemporaryArray<StringIndentationRegion>.Empty;
 
         using var _ = ArrayBuilder<SyntaxNode>.GetInstance(out var nodeStack);
         nodeStack.Add(root);
 
         while (nodeStack.TryPop(out var node))
         {
-            // DoNot' bother recursing into nodes that don't hit the requested span, they can never contribute 
-            // regions of interest.
-            if (!node.Span.IntersectsWith(textSpan))
```
Comment (ToddGrun) [inline src/Features/CSharp/Portable/StringIndentation/CSharpStringIndentationService.cs:59]: >if (!node.Span.IntersectsWith(textSpan)) [](http://example.com/codeflow?start=12&length=40)

seems like we did this inefficiently in a lot of places.

Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/StringIndentation/CSharpStringIndentationService.cs:59]: Yup!  Tackling the ones that show up in traces first

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**